### PR TITLE
Update list of "ARM-based ProMicro replacements"

### PR DESCRIPTION
**Pull request by PancakeGoon**
Added Boardsource's upcoming 'Blok' devboard to the reference-list of ARM-based ProMicro alternatives, within the "Firmware Squeezing" docs for users needing more firmware space. This being an especially helpful addition right now, while it's still in "Pre-Sale" at a discounted price for a short-time.

## Description

Updated the list of ARM-based development boards (listed at the bottom of the how-to Docs for 'Squeezing Firmware Space") with Pro-Micro PCB footprints to include a new option available for purchase (listed at the bottom of the QMK Docs for 'Squeezing Firmware Space").

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

File: docs/squeezing_avr.md [hunk 1]
```diff
@@ -189,6 +189,7 @@ That said, there are a number of Pro Micro replacements with ARM controllers:
 * [Proton C](https://qmk.fm/proton-c/) (out of stock)
 * [Bonsai C](https://github.com/customMK/Bonsai-C) (Open Source, DIY/PCBA)
 * [Raspberry Pi 2040](https://www.sparkfun.com/products/18288) (not currently supported, no ETA)
```
Comment (filterpaper) [inline docs/squeezing_avr.md:191]:
```suggestion
* [Adafruit KB2040](https://learn.adafruit.com/adafruit-kb2040)
* [SparkFun Pro Micro - RP2040](https://www.sparkfun.com/products/18288)
```

These should be updated together with Blok as all 3 RP2040 boards will be supported in the coming merge cycle.

Comment (keyboard-magpie) [reply docs/squeezing_avr.md:191]: Fully agree.

Comment (PancakeGoon) [reply docs/squeezing_avr.md:191]: > ```suggestion
> * [STeMCell](https://github.com/megamind4089/STeMCell) (Open Source, DIY/PCBA)
> * [Adafruit KB2040](https://learn.adafruit.com/adafruit-kb2040)
> * [SparkFun Pro Micro - RP2040](https://www.sparkfun.com/products/18288)
> ```
> 
> These should be updated together with Blok as all 3 RP2040 boards and the STM controller will be supported in the August merge cycle.

Nice, thanks! 
Okay, I'm still getting the hang of all this, so correct me if I'm wrong; but if I commit your edit to my PR;
- this is just updating/"replacing" the _old_ "RasPi 2040 (not supported, ... )"-entry by removing the incompatibility warning, as well as adding those additional 2 devboards to the list?
- but everything originally listed before-&-after this edit—including the Blok addition—will be kept? 

Basically, it appears to me that your edit was you just taking this opportunity to further-update _another_ entry on the list, _on-top of_ my original Docs addition, in-order to keep it all together in one PR commit - and I just want to make sure I'm understanding this correctly before I go ahead, so I can better understand this aspect of contributing & PR's. 🙃

Comment (filterpaper) [reply docs/squeezing_avr.md:191]: Your understanding is correct. The three green lines are showing what will be added if you commit the suggested changes. Your blok addition will remain unchanged below them.

Comment (filterpaper) [reply docs/squeezing_avr.md:191]: Boardsource just announced that they will start shipping so the prefix `(currently in Pre-Order phase, estimated In-Stock "August 2022")` should be removed now.

Comment (PancakeGoon) [reply docs/squeezing_avr.md:191]: Gotcha, will do & thanks!

File: docs/squeezing_avr.md [hunk 2]
```diff
@@ -188,7 +188,10 @@ If you've done all of this, and your firmware is still too large, then it's time
 That said, there are a number of Pro Micro replacements with ARM controllers: 
 * [Proton C](https://qmk.fm/proton-c/) (out of stock)
 * [Bonsai C](https://github.com/customMK/Bonsai-C) (Open Source, DIY/PCBA)
-* [Raspberry Pi 2040](https://www.sparkfun.com/products/18288) (not currently supported, no ETA)
+* [STeMCell](https://github.com/megamind4089/STeMCell) (Open Source, DIY/PCBA)
+* [Adafruit KB2040](https://learn.adafruit.com/adafruit-kb2040)
+* [SparkFun Pro Micro - RP2040](https://www.sparkfun.com/products/18288)
+* [Blok](https://boardsource.xyz/store/628b95b494dfa308a6581622) (currently in Pre-Order phase, estimated In-Stock "August 2022")
```
Comment (filterpaper) [inline docs/squeezing_avr.md:194]:
```suggestion
* [Blok](https://boardsource.xyz/store/628b95b494dfa308a6581622)
```

The suffix is moot because they are shipping now.

**Review by PancakeGoon — COMMENTED** [review]
Removed the "Pre-order & stock estimate for Blok units, as Boardsource announced that the manufacturer has now finished production, and for all intents & purposes they're In-Stock.

**Review by filterpaper — APPROVED** [review]
Thanks for the updates!

**Review by zvecr — APPROVED** [review]

**Comment by PancakeGoon** [comment]
Thanks y'all - successful first contribute, say word! 🎉
